.github/scripts/get_workflow_job_id.py

```diff
@@ -37,16 +38,20 @@ def fetch_url(url: str, *,
               reader: Callable[[Any], Any] = lambda x: x.read()) -> Any:
     if headers is None:
         headers = {}
-    try:
-        with urlopen(Request(url, headers=headers)) as conn:
-            return reader(conn)
-    except urllib.error.HTTPError as err:
-        exception_message = (
-            "Is github alright?",
-            f"Recieved status code '{err.code}' when attempting to retrieve {url}:\n",
-            f"{err.reason}\n\nheaders={err.headers}"
-        )
-        raise RuntimeError(exception_message) from err
+    retries = 3
+    for i in range(retries + 1):
+        try:
```

Comment:
Sorry for the late review, but this change does two things which are imo bad from design point of view:
 - Hardcodes constants
 - Builds [Pyramid of Doom](https://en.wikipedia.org/wiki/Pyramid_of_doom_(programming))
Let me propose a PR that fixes the problem

Reply:
https://github.com/pytorch/pytorch/pull/93874